Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -884,7 +884,7 @@ bool RenderLayer::canRender3DTransforms() const
 
 bool RenderLayer::paintsWithFilters() const
 {
-    if (!renderer().hasFilter())
+    if (!hasFilter() || (hasFilter() && renderer().isSVGRootOrLegacySVGRoot()))
```

Comment:
You do not have to check for `hasFilter()` in the second clause:
```
if (!hasFilter() || renderer().isSVGRootOrLegacySVGRoot())
```
This is because `A || !AB <=> A || B`

Or you can add the new statement:
```
if (renderer().isSVGRootOrLegacySVGRoot())
    return false;
```